lib/srv/db/ca.go

```diff
@@ -205,54 +317,98 @@ func (d *realDownloader) Download(ctx context.Context, database types.Database,
 	case types.DatabaseTypeAWSKeyspaces:
 		return d.downloadFromURL(amazonKeyspacesCAURL)
 	}
-	return nil, trace.BadParameter("%v doesn't support automatic CA download", database)
+	return nil, nil, trace.BadParameter("%v doesn't support automatic CA download", database)
+}
+
+// GetVersion returns the CA version for the provided database.
+func (d *realDownloader) GetVersion(ctx context.Context, database types.Database, hint string) ([]byte, error) {
+	switch database.GetType() {
+	case types.DatabaseTypeRDS:
+		return d.getVersionFromURL(database, rdsCAURLForDatabase(database))
+	case types.DatabaseTypeRedshift:
+		return d.getVersionFromURL(database, redshiftCAURLForDatabase(database))
+	case types.DatabaseTypeElastiCache,
+		types.DatabaseTypeMemoryDB:
+		return d.getVersionFromURL(database, amazonRootCA1URL)
+	case types.DatabaseTypeAzure:
+		if strings.HasSuffix(azureCAURLBaltimore, hint) {
+			return d.getVersionFromURL(database, azureCAURLBaltimore)
+		} else if strings.HasSuffix(azureCAURLDigiCert, hint) {
+			return d.getVersionFromURL(database, azureCAURLDigiCert)
+		}
+		return nil, trace.BadParameter("unknown Azure CA %q", hint)
+	case types.DatabaseTypeAWSKeyspaces:
+		return d.getVersionFromURL(database, amazonKeyspacesCAURL)
+	}
+
+	return nil, trace.NotImplemented("%v doesn't support fetching CA version", database)
 }
 
 // downloadFromURL downloads root certificate from the provided URL.
-func (d *realDownloader) downloadFromURL(downloadURL string) ([]byte, error) {
-	resp, err := http.Get(downloadURL)
+func (d *realDownloader) downloadFromURL(downloadURL string) ([]byte, []byte, error) {
+	resp, err := d.httpClient.Get(downloadURL)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, nil, trace.Wrap(err)
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return nil, trace.BadParameter("status code %v when fetching from %q",
+		return nil, nil, trace.BadParameter("status code %v when fetching from %q",
 			resp.StatusCode, downloadURL)
 	}
 	bytes, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return nil, nil, trace.Wrap(err)
 	}
-	return bytes, nil
+
+	// If there is a http response and it does have the "etag" header present,
+	// use it as the CA version.
+	if resp.Header.Get("ETag") != "" {
+		return bytes, []byte(resp.Header.Get("ETag")), nil
+	}
+
+	// Otherwise, hash the contents and return it as the version.
+	hash := sha256.Sum256(bytes)
+	return bytes, hash[:], nil
 }
 
 // downloadForCloudSQL downloads root certificate for the provided Cloud SQL
 // instance.
 //
 // This database service GCP IAM role should have "cloudsql.instances.get"
 // permission in order for this to work.
-func (d *realDownloader) downloadForCloudSQL(ctx context.Context, database types.Database) ([]byte, error) {
-	sqladminService, err := sqladmin.NewService(ctx)
+func (d *realDownloader) downloadForCloudSQL(ctx context.Context, database types.Database) ([]byte, []byte, error) {
```

Comment:
Can version be a `string` instead of `[]byte`?

Reply:
Do you mean the CloudSQL version or the `GetVersion` return? As for the `GetVersion` return, I don't see any advantage besides reducing casting. Although we would require casting in the `sha256` result and when writing into the file. Are you considering any other advantage (like placing it in logs or somewhere)?

Reply:
> Are you considering any other advantage (like placing it in logs or somewhere)?
Nope =p. Just personally feel `version` is more of a string type. really don't mind either way.